src/durmod/kaldi-durmod.cc

```diff
@@ -0,0 +1,463 @@
+      if (binary_feats_.count(phone) <= 0) {
+        KALDI_WARN << "Phone " << phone << " does not have any "
+                   << "acoustic question associated.";
+      }
+    }
+  }
+}
+
+void PhoneDurationModel::InitNnet(int input_dim, int dim1,
+                                  int dim2, int output_dim) {
+  std::stringstream config;
+
+  // TODO(hhadian): to be later moved to scripts:
+
+  KALDI_LOG << "DurModel.Nnet: in-dim: " << input_dim
+            << ", dim1: " << dim1
+            << ", dim2: " << dim2
+            << ", out-dim: " << output_dim;
+
+  config << "component name=affine1 type=AffineComponent"
+         << " learning-rate=0.001 param-stddev=0.02 bias-stddev=0"
+         << " input-dim=" << input_dim << " output-dim=" << dim1 << std::endl;
+  config << "component name=relu1 type=RectifiedLinearComponent"
+         << " dim=" << dim1 << std::endl;
+  config << "component name=norm1 type=NormalizeComponent"
+         << " dim=" << dim1 << std::endl;
+  config << "component name=affine2 type=AffineComponent"
+         << " learning-rate=0.001 param-stddev=0.02 bias-stddev=0"
+         << " input-dim=" << dim1 << " output-dim=" << dim2 << std::endl;
+  config << "component name=relu2 type=RectifiedLinearComponent"
+         << " dim=" << dim2 << std::endl;
+  config << "component name=norm2 type=NormalizeComponent"
+         << " dim=" << dim2 << std::endl;
+  config << "component name=affine3 type=AffineComponent"
+         << " learning-rate=0.001 param-stddev=0.02 bias-stddev=0"
+         << " input-dim=" << dim2 << " output-dim=" << output_dim << std::endl;
+  config << "component name=softmax type=LogSoftmaxComponent"
+         << " dim=" << output_dim << std::endl;
+  config << "input-node name=input dim=" << input_dim << std::endl;
+  config << "component-node name=affine1_node component=affine1 input=input"
+         << std::endl;
+  config << "component-node name=relu1_node component=relu1 input=affine1_node"
+         << std::endl;
+  config << "component-node name=norm1_node component=norm1 input=relu1_node"
+         << std::endl;
+  config << "component-node name=affine2_node "
+         << "component=affine2 input=norm1_node"
+         << std::endl;
+  config << "component-node name=relu2_node component=relu2 input=affine2_node"
+         << std::endl;
+  config << "component-node name=norm2_node component=norm2 input=relu2_node"
+         << std::endl;
+  config << "component-node name=affine3_node "
+         << "component=affine3 input=norm2_node"
+         << std::endl;
+  config << "component-node name=softmax_node "
+         << "component=softmax input=affine3_node"
+         << std::endl;
+  config << "output-node name=output input=softmax_node" << std::endl;
+  std::cout << config.str();
+  nnet_.ReadConfig(config);
+}
+
+void PhoneDurationModel::Read(std::istream &is, bool binary) {
+  ExpectToken(is, binary, "<PhoneDurationModel>");
+  ReadBasicType(is, binary, &left_context_);
```

Comment:
for readability in text form, probably add separate tags 
   <LeftContext>, <RightContext> and <MaxDuration>.